[IMP] purchase: split purchase agreements docs

brse-odoo opened the pull request:
This PR is targeting Odoo project task #3051002 to update the Calls for Tenders portion of the Purchase Agreements doc.

Performed complete rewrite for Purchase document: **Purchase agreements: blanket orders and calls for tenders.**
- Split blanket orders and calls for tenders into 2 separate .rst files
- Completely rewrote each .rst file from scratch and added example flows to each
- For calls for tenders doc: completely revamped for v16, made v16-specific changes
- Replaced and updated all screenshots
- Rearranged toc file tree for new docs and files

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/documentation/pull/3511).

brse-odoo commented:
Hi @ksc-odoo -- when you have a chance, can you please review this doc? Thanks in advance for your help on this!

ksc-odoo commented on content/applications/inventory_and_mrp/purchase/manage_deals/blanket_orders.rst:
```diff
@@ -0,0 +1,98 @@
+================================
+
+Once a blanket order is confirmed, a new vendor line is added under the :guilabel:`Purchase` tab of
+of the products included in the :abbr:`BO (blanket order)`. This makes blanket orders useful when
+it comes to :doc:`automated replenishment <../../purchase/products/reordering>`, because
```
This makes blanket orders useful with :doc:`automated replenishment <../../purchase/products/reordering>` because...

ksc-odoo commented on content/applications/inventory_and_mrp/purchase/manage_deals/blanket_orders.rst:
```diff
@@ -0,0 +1,98 @@
+of the products included in the :abbr:`BO (blanket order)`. This makes blanket orders useful when
+it comes to :doc:`automated replenishment <../../purchase/products/reordering>`, because
+information about the :guilabel:`Vendor`, :guilabel:`Price`, and the :guilabel:`Agreement`
+referenced on the vendor line can be used to determine where, when, and for what price this
+product could be replenished for.
```
...referenced on the vendor line. This information is used to determine where, when, and for what price this
product could be replenished.

ksc-odoo commented on content/applications/inventory_and_mrp/purchase/manage_deals/calls_for_tenders.rst:
```diff
@@ -0,0 +1,139 @@
+Calls for Tenders
+=================
+
+**Calls for Tenders** are one of the two main types of **Purchase Agreements** available by default
+in Odoo.
```
The **Calls for Tenders** type of **Purchase Agreements** is available by default in Odoo.

(the initial wording seemed a bit clunky to me, so the above is just a suggestion. Nothing required - just a thought)

ksc-odoo commented on content/applications/inventory_and_mrp/purchase/manage_deals/calls_for_tenders.rst:
```diff
@@ -0,0 +1,139 @@
+in Odoo.
+
+*Calls for Tenders* are used to request offers from multiple vendors at the same time, by inviting
+those vendors to submit offers for similar goods or services all at once. This helps companies that
```
...(or services)...

ksc-odoo commented on content/applications/inventory_and_mrp/purchase/manage_deals/calls_for_tenders.rst:
```diff
@@ -0,0 +1,139 @@
+
+*Calls for Tenders* are used to request offers from multiple vendors at the same time, by inviting
+those vendors to submit offers for similar goods or services all at once. This helps companies that
+use this feature to select the cheapest and fastest vendors in the market, depending on their
+needs.
+
```
...use this feature to select the cheapest (and fastest) vendors, depending on their specific business needs.

ksc-odoo commented on content/applications/inventory_and_mrp/purchase/manage_deals/calls_for_tenders.rst:
```diff
@@ -0,0 +1,139 @@
+use this feature to select the cheapest and fastest vendors in the market, depending on their
+needs.
+
+While calls for tenders are primaily used by organizations in the public sector who are legally
```
Calls for Tenders...like the other features I've commented on above, should look uniform throughout the doc - and they should stand out in a sentence, to make things a bit clearer for the reader. So, either capitalize it throughout, toss some * on either side, hit it with a :guilabel: ... someting like that. What do you think?

ksc-odoo replied:
...in the public sector, who are legally...

brse-odoo replied:
I definitely agree with you on making it uniform and stand our throughout the doc. After first mention, I've decided to capitalize.

ksc-odoo commented on content/applications/inventory_and_mrp/purchase/manage_deals/calls_for_tenders.rst:
```diff
@@ -0,0 +1,139 @@
+needs.
+
+While calls for tenders are primaily used by organizations in the public sector who are legally
+bound to use them when making a purchase, private companies can also use them to spend efficiently,
+as well.
```
...private companies can also use them to spend money efficiently, as well.

ksc-odoo commented on content/applications/inventory_and_mrp/purchase/manage_deals/calls_for_tenders.rst:
```diff
@@ -0,0 +1,139 @@
+bound to use them when making a purchase, private companies can also use them to spend efficiently,
+as well.
+
+When multiple offers are sent to a company, they can pick the one that works best for them,
+minimizing costs and increasing profitability in the most efficient manner.
```
When multiple offers are sent to a company, the ideal/best option can be chosen - minimizing costs and increasing profitability in the most efficient manner.

ksc-odoo commented on content/applications/inventory_and_mrp/purchase/manage_deals/calls_for_tenders.rst:
```diff
@@ -0,0 +1,139 @@
+    To save time on a call for tender, custom vendors, prices, and delivery lead times can be set
+    in the :guilabel:`Purchase` tab on a product form.
+
+Create a :abbr:`RFQ (request for quotation)`
+--------------------------------------------
```
Create an :abbr:`RFQ (request for quotation)`
-------------------------------------------------

ksc-odoo commented on content/applications/inventory_and_mrp/purchase/manage_deals/calls_for_tenders.rst:
```diff
@@ -0,0 +1,139 @@
+Create a :abbr:`RFQ (request for quotation)`
+--------------------------------------------
+
+From the Purchase main dashboard, click :guilabel:`New` to create a new request for quotation.
```
...the main Purchase dashboard...

ksc-odoo commented on content/applications/inventory_and_mrp/purchase/manage_deals/calls_for_tenders.rst:
```diff
@@ -0,0 +1,139 @@
+Then, fill out the information, and add products to the product lines. Once all the information is
+entered, click :guilabel:`Send by Email`. This can be useful when creating a call for tender,
+because vendors can confirm if their past prices still hold today, which can help companies choose
+the best offers for them.
```
...the best offers.

ksc-odoo commented on content/applications/inventory_and_mrp/purchase/manage_deals/calls_for_tenders.rst:
```diff
@@ -0,0 +1,139 @@
+because vendors can confirm if their past prices still hold today, which can help companies choose
+the best offers for them.
+
+Create alternatives to a :abbr:`RFQ (request for quotation)`
+------------------------------------------------------------
```
Create alternatives to an :abbr:`RFQ (request for quotation)`
------------------------------------------------------------------

ksc-odoo commented on content/applications/inventory_and_mrp/purchase/manage_deals/calls_for_tenders.rst:
```diff
@@ -0,0 +1,139 @@
+Create alternatives to a :abbr:`RFQ (request for quotation)`
+------------------------------------------------------------
+
+Once a purchase order is created, alternative quotations can be created and sent to vendors to
```
...created (and sent)...

ksc-odoo commented on content/applications/inventory_and_mrp/purchase/manage_deals/calls_for_tenders.rst:
```diff
@@ -0,0 +1,139 @@
+------------------------------------------------------------
+
+Once a purchase order is created, alternative quotations can be created and sent to vendors to
+compare for the best deal later. To do this, click the :guilabel:`Alternatives` tab, then click
```
...to compare late.

ksc-odoo commented on content/applications/inventory_and_mrp/purchase/manage_deals/calls_for_tenders.rst:
```diff
@@ -0,0 +1,139 @@
+
+Once a purchase order is created, alternative quotations can be created and sent to vendors to
+compare for the best deal later. To do this, click the :guilabel:`Alternatives` tab, then click
+:guilabel:`Create Alternative`. This will cause a popup to appear.
```
...When clicked, a pop-up appears.

ksc-odoo commented on content/applications/inventory_and_mrp/purchase/manage_deals/calls_for_tenders.rst:
```diff
@@ -0,0 +1,139 @@
+alternative :abbr:`PO (purchase order)`. Next to this, there is a :guilabel:`Copy Products` checkbox
+that is selected by default. When selected, the product quantities of the original
+:abbr:`PO (purchase order)` will be copied to the alternative. Once finished, click
+:guilabel:`Create Alternative`. This will create and navigate to a new purchase order.
```
...This creates (and navigates) to a new purchase order.

ksc-odoo commented on content/applications/inventory_and_mrp/purchase/manage_deals/calls_for_tenders.rst:
```diff
@@ -0,0 +1,139 @@
+this time, *uncheck* the :guilabel:`Copy Products` checkbox. Click :guilabel:`Create Alternative`.
+
+.. tip::
+    If one or two alternative quotations need to be removed from the :guilabel:`Alternatives` tab,
```
If one (or two)...

ksc-odoo commented on content/applications/inventory_and_mrp/purchase/manage_deals/calls_for_tenders.rst:
```diff
@@ -0,0 +1,139 @@
+    If one or two alternative quotations need to be removed from the :guilabel:`Alternatives` tab,
+    they can be individually removed by clicking on the black "X" icon at the end of their row.
+
+This will create a new purchase order, but since the product quantities of the original
```
This creates...

ksc-odoo commented on content/applications/inventory_and_mrp/purchase/manage_deals/calls_for_tenders.rst:
```diff
@@ -0,0 +1,139 @@
+
+This will create a new purchase order, but since the product quantities of the original
+:abbr:`PO (purchase order)` were not copied over, new products will need to be added to the product
+lines. Add the products, then click :guilabel:`Send by Email`.
```
...After the products are added, click :guilabel:`Send by Email`.

ksc-odoo commented on content/applications/inventory_and_mrp/purchase/manage_deals/calls_for_tenders.rst:
```diff
@@ -0,0 +1,139 @@
+:abbr:`PO (purchase order)` were not copied over, new products will need to be added to the product
+lines. Add the products, then click :guilabel:`Send by Email`.
+
+From this newest purchase order, click the :guilabel:`Alternatives` tab. From here, all three
+purchase orders can be seen, inluding which vendors they were sent to, the expected arrival of the
+products if confirmed, and the status of the order. Before comparing any product lines, click
+:guilabel:`New` at the top of the page. This will create a brand new quotation. Fill out the
+information on this new quotation, and click :guilabel:`Send by Email`. Then, click the
+:guilabel:`Alternatives` tab once more. On this new :abbr:`PO (purchase order)`, there are no other
+orders linked. To link this order with the alternatives, click :guilabel:`Link to Existing RfQ`.
```
From this newest purchase order, click the :guilabel:`Alternatives` tab. 

From here, all three purchase orders can be seen, including which vendors they were sent to, the expected arrival of the products (if confirmed), and the status of the order. 

Before comparing any product lines, click :guilabel:`New` at the top of the page. This will create a brand new quotation. Fill out the information on this new quotation, and click :guilabel:`Send by Email`. Then, click the
:guilabel:`Alternatives` tab once more. On this new :abbr:`PO (purchase order)`, there are no other
orders linked. To link this order with the alternatives, click :guilabel:`Link to Existing RfQ`.

ksc-odoo commented on content/applications/inventory_and_mrp/purchase/manage_deals/calls_for_tenders.rst:
```diff
@@ -0,0 +1,139 @@
+   :align: center
+   :alt: Popup to link new quotation to existing RFQs.
+
+This will cause a popup to appear. From the popup, select the three purchase orders created
```
This causes a pop-up to appear.

(you can use popup or pop-up - don't think that matters. But, whichever you choose, make sure it stays uniform throughout the documentation)

ksc-odoo commented on content/applications/inventory_and_mrp/purchase/manage_deals/calls_for_tenders.rst:
```diff
@@ -0,0 +1,139 @@
+Compare product lines
+---------------------
+
+Now that there are multiple RFQs linked as alternatives, they can be compared to determine which
```
...compared in order to determine...

ksc-odoo commented on content/applications/inventory_and_mrp/purchase/manage_deals/calls_for_tenders.rst:
```diff
@@ -0,0 +1,139 @@
+
+From the *Compare Order Lines* page, each product included in any of the purchase orders will be
+shown in a drop-down, with all of the orders they were included in listed below them. In the
+different columns, the quantities, unit price, and total price of the products can be compared. At
+the end of each row, products can be selected by clicking :guilabel:`Choose`. Once all the desired
+products have been chosen, click :guilabel:`Requests for Quotation` in the breadcrumbs at the top of
+the page to navigate back to an overview of all RFQs.
```
different columns, the quantities, unit price, and total price of the products can be compared. 

At the end of each row, products can be selected by clicking :guilabel:`Choose`. Once all the desired
products have been chosen, click :guilabel:`Requests for Quotation` (in the breadcrumbs, at the top of
the page) to navigate back to an overview of all RFQs.

ksc-odoo commented on content/applications/inventory_and_mrp/purchase/manage_deals/calls_for_tenders.rst:
```diff
@@ -0,0 +1,139 @@
+the page to navigate back to an overview of all RFQs.
+
+.. note::
+    To remove one or two product lines from the *Compare Order Lines* page, click :guilabel:`Clear`
```
...one (or two)...

ksc-odoo commented on content/applications/inventory_and_mrp/purchase/manage_deals/calls_for_tenders.rst:
```diff
@@ -0,0 +1,139 @@
+    To remove one or two product lines from the *Compare Order Lines* page, click :guilabel:`Clear`
+    at the end of that product line's row.
+
+Cancel or keep alternatives
+---------------------------
```
Cancel (or keep) alternatives
-------------------------------

ksc-odoo commented on content/applications/inventory_and_mrp/purchase/manage_deals/calls_for_tenders.rst:
```diff
@@ -0,0 +1,139 @@
+Cancel or keep alternatives
+---------------------------
+
+Now that the desired products have been chosen based on which vendors provided the best offer, the
```
have been chosen, based on which vendors provided the best offer, the...

ksc-odoo commented on content/applications/inventory_and_mrp/purchase/manage_deals/calls_for_tenders.rst:
```diff
@@ -0,0 +1,139 @@
+---------------------------
+
+Now that the desired products have been chosen based on which vendors provided the best offer, the
+other RFQs can be canceled. Under the *Total* column, the RFQs from which no products were chosen
```
...the RFQs (from which no products were chosen)...

ksc-odoo commented on content/applications/inventory_and_mrp/purchase/manage_deals/calls_for_tenders.rst:
```diff
@@ -0,0 +1,139 @@
+other RFQs can be canceled. Under the *Total* column, the RFQs from which no products were chosen
+have automatically had their total cost set to 0.
+
+To confirm the quotations that contain the chosen product quantities, click into them, and click
```
...click into them, and select...

ksc-odoo commented on content/applications/inventory_and_mrp/purchase/manage_deals/calls_for_tenders.rst:
```diff
@@ -0,0 +1,139 @@
+have automatically had their total cost set to 0.
+
+To confirm the quotations that contain the chosen product quantities, click into them, and click
+:guilabel:`Confirm Order`. This will cause an *Alternative Warning* popup to appear. From this
```
...This causes an...

ksc-odoo commented on content/applications/inventory_and_mrp/purchase/manage_deals/calls_for_tenders.rst:
```diff
@@ -0,0 +1,139 @@
+   :align: center
+   :alt: Keep or cancel popup for alternative RFQs.
+
+:guilabel:`Cancel Alternatives` will automatically cancel the alternative purchase orders.
```
...automatically cancels...

ksc-odoo commented on content/applications/inventory_and_mrp/purchase/manage_deals/calls_for_tenders.rst:
```diff
@@ -0,0 +1,139 @@
+   :alt: Keep or cancel popup for alternative RFQs.
+
+:guilabel:`Cancel Alternatives` will automatically cancel the alternative purchase orders.
+:guilabel:`Keep Alternatives` will keep the alternative purchase orders open, so they can still be
```
...keeps the alternative...

ksc-odoo commented on content/applications/inventory_and_mrp/purchase/manage_deals/calls_for_tenders.rst:
```diff
@@ -0,0 +1,139 @@
+accessed if any additional product quantities need to be ordered. Once all products are ordered,
+:guilabel:`Cancel Alternatives` can be selected from whichever :abbr:`PO (purchase order)` is open.
+
+Once all product quantities have been ordered, the purchase process can be followed and continued
```
...can be followed, and continued...

ksc-odoo reviewed (commented):
Hey there. Great job. Just made some minor grammatical adjustments here and there. But, once those small issues are addressed, I think it'll be ready for the next level.

brse-odoo commented on content/applications/inventory_and_mrp/purchase/manage_deals/blanket_orders.rst:
```diff
@@ -0,0 +1,98 @@
+they can be advantageous when negotiating bulk pricing with vendors.
+
+.. note::
+   To create and use blanket orders, *Purchase Agreements* must be activated by going to
```
Nice catch, definitely a good call to use the `:guilabel:` tag for terms like this.

brse-odoo commented on content/applications/inventory_and_mrp/purchase/manage_deals/blanket_orders.rst:
```diff
@@ -0,0 +1,98 @@
+After confirming a blanket order, new quotations can be created that will be linked to the blanket
+order. To create a new :abbr:`RFQ (request for quotation)`, click :guilabel:`New Quotation`. Then,
+set the :guilabel:`Quantity` and :guilabel:`Unit Price` on the product lines. Once finished, click
+:guilabel:`Confirm Order` to create a purchase order. Click back to the blanket order via the
+breadcrumbs, and repeat to create as many new quotations as needed.
```
That's a great suggestion, I incorporated this last part as a tip!

StraubCreative commented on content/applications/inventory_and_mrp/purchase/manage_deals/calls_for_tenders.rst:
```diff
@@ -0,0 +1,144 @@
+=================
+Calls for Tenders
+=================
```
Sentence case; this is not a proper noun.
```suggestion
=================
Calls for tenders
=================
```

StraubCreative commented on content/applications/inventory_and_mrp/purchase/manage_deals/calls_for_tenders.rst:
```diff
@@ -0,0 +1,144 @@
+Calls for Tenders
+=================
+
+*Calls for Tenders* are one of the two main types of :guilabel:`Purchase Agreements` available by
+default in Odoo.
```
No guilabel here
```suggestion
*Calls for Tenders* are one of the two main types of purchase agreements available by default in
Odoo.
```

StraubCreative reviewed (changes requested):
Hi @brse-odoo 

Pausing here. Can you take another look at thes docs and be sure to:
- use sentence case (calls for tenders is not a proper noun)
- be sure each heading is concise (don't need "Example flow:" f/e)
- use guilabels appropriately
- remove vague/unnecessary language
- break 100th char lines correctly (I think your linter is off by 1 character)
- etc.

I'll be pushing up an updated commit which has the merge conflict solved, a fresh rebase, and a couple items already fixed on `calls_for_tenders.rst`. Can pick it up from there :)

Thanks and please click **Re-request review** (the 🔄 button) when it's ready for another look.

brse-odoo commented:
Hi @thomasdeleval, this PR has been marked as high priority and ZST and I are working to merge it by the the 4/27 deadline (EOW).

I have tagged you here for review if you would like to review for content by EOD 4/26, so we can keep the timeline and merge on time.

Thank you, and let me know if you have any questions!

jcs-odoo commented on content/applications/inventory_and_mrp/purchase/manage_deals/blanket_orders.rst:
```diff
@@ -0,0 +1,203 @@
+Create a new blanket order
+==========================
+
+To create blanker orders, the *Purchase Agreements* feature first needs to be enabled in the
+settings of the :menuselection:`Purchase` app. To do this, go to :menuselection:`Purchase -->
+Configuration --> Settings`, and under the :guilabel:`Orders` section, click the
+:guilabel:`checkbox` next to :guilabel:`Purchase Agreements`. Doing so will enable the ability to
+create blanket orders, as well as alternative :abbr:`RFQs (Requests for Quotation)`.
```
I think some paragraphs can be more concise. This is just an example:

```suggestion
To create blanker orders and alternative :abbr:`RFQs (Requests for Quotation),
 go to :menuselection:`Purchase -->
Configuration --> Settings`, and
enable  :guilabel:`Purchase Agreements`. 
```

jcs-odoo commented on content/applications/inventory_and_mrp/purchase/manage_deals/blanket_orders.rst:
```diff
@@ -0,0 +1,203 @@
+   :align: center
+   :alt: New blanket order purchase agreement with added products.
+
+Once all relevant fields have been filled out, click :guilabel:`Add a line` to add products to the
+product lines under the :guilabel:`Product` column. Then, change the quantity of each product in
+the :guilabel:`Quantity` column (if desired), and set a price in the :guilabel:`Unit Price` column.
```
While it's correct, I think you can just describe this kind of possibilities, as it is quite obvious:
```suggestion
Once all relevant fields have been filled out, click :guilabel:`Add a line` to add products,
then, set the quantity
and unit price.
```

jcs-odoo reviewed (commented):
Hello!

You asked me to give some feedback, but I'm afraid I could barely touch the surface of all your work, and I didn't have the time to read it thoroughly. Anyway, here are some comments:

- see my comment in the diff about the split of the agreement page and the implication within the app's settings. It's mostly a question for Thomas.
- the titles seem long. It's best to keep them concise to ease the navigation in the documentation website. [See guidelines about headings and titles](https://www.odoo.com/documentation/master/contributing/documentation/content_guidelines.html#titles-and-headings)
- Similarly, I think some paragraphs can be more concise (see some examples in the diff)

I hope it helps, but it's great that the docs will soon be available to the user.

Have a good day!

brse-odoo commented:
Made small changes/fixes to rst:
- Removed `:guilabels:` from all **forms**
- Added `:seealso:` tags at top of each doc to reference other doc
- Fixed 100th character breaks

@StraubCreative **This is ready for your final review.** Thank you!

StraubCreative reviewed (changes requested):
Hi @brse-odoo 
Some feedbacks below, mostly around formatting. The content looks good 👍 
Will be pushing up changes momentarily and then merging.

StraubCreative reviewed (approved):
@robodoo r+

jcs-odoo commented on content/applications/inventory_and_mrp/purchase/manage_deals.rst:
```diff
@@ -7,6 +7,7 @@ Manage deals
    :titlesonly:
 
-   manage_deals/agreements
+   manage_deals/blanket_orders
+   manage_deals/calls_for_tenders
```
Cool :)
It's maybe best to leave feedback in the project RD Feedback so @thomasdeleval can decide.
